gateleen-routing/src/main/java/org/swisspush/gateleen/routing/CustomHttpResponseHandler.java

```diff
@@ -0,0 +1,44 @@
+package org.swisspush.gateleen.routing;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.http.HttpServerRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handler to respond with a custom status code
+ * @author henningo
+ */
+public class CustomHttpResponseHandler {
+
+    private final Logger LOG = LoggerFactory.getLogger(CustomHttpResponseHandler.class);
+
+    private final String path;
+
+    public CustomHttpResponseHandler(String rootPath) {
+        this.path = rootPath;
+        LOG.info("listening to {}", rootPath);
+    }
+
+    public boolean handle(HttpServerRequest request) {
+        if (!request.uri().startsWith(path)) {
+            return false;
+        }
+        HttpResponseStatus rs;
+        String info = "";
+        try {
+            String code = request.uri().substring(path.length());
```

Comment:
@hiddenalpha proposed to isolate the "numeric" part of the URI-trailer, i.e. to also accept
`/gateleen/server/return-with-status-code/404/some/other/path/segments/which/are/ignored`

This gives us the option to work woth a Route-Hook where _destination is this Handler_.
Note that Route-Hooks always add the Path-Segments 'right of the hooked path' to the destination-URL